native/src/main/java/io/ballerina/stdlib/http/api/client/actions/HttpClientAction.java

```diff
@@ -293,22 +301,69 @@ private static String[] getPathStringArray(BArray pathArray) {
 
     private static String constructQueryString(BMap params) {
         List<String> queryParams = new ArrayList<>();
+        Map<String, String> annotationValues = getQueryNameMapping(params);
         BString[] keys = (BString[]) params.getKeys();
         if (keys.length == 0) {
             return "";
         }
         for (BString key : keys) {
             Object value = params.get(key);
+            String queryName = key.getValue();
+            queryName = annotationValues.getOrDefault(queryName, queryName);
             String valueString = value.toString();
             if (value instanceof BArray) {
                 valueString = valueString.substring(1, valueString.length() - 1);
                 valueString = valueString.replace(QUOTATION_MARK, EMPTY);
             }
-            queryParams.add(key.getValue() + EQUAL_SIGN + valueString);
+            queryParams.add(queryName + EQUAL_SIGN + valueString);
         }
         return String.join(AND_SIGN, queryParams);
     }
 
+    /**
+     * This util function extracts the query name with the query annotation.
+     *
+     * @param params - Parameter map
+     * @return Map of string with overridden query param names
+     */
+    private static Map<String, String> getQueryNameMapping(BMap params) {
+        Map<String, String> annotationValues = new HashMap<>();
+        if (!(params.getType() instanceof RecordType queryRecord)) {
+            return annotationValues;
+        }
+        BMap<BString, Object> queryFields = queryRecord.getAnnotations();
+
+        for (Map.Entry<BString, Object> qField: queryFields.entrySet()) {
+            BMap value = (BMap) qField.getValue();
+            Object[] keys = value.getKeys();
+            for (Object annotRef: keys) {
+                String refRegex = ModuleUtils.getHttpPackageIdentifier() + COLON + ANN_NAME_QUERY;
+                Pattern pattern = Pattern.compile(refRegex);
+                Matcher matcher = pattern.matcher(annotRef.toString());
+                if (matcher.find()) {
+                    BMap refValue = (BMap) value.get(annotRef);
+                    extractedFieldName(annotationValues, qField, refValue);
+                }
+            }
+        }
+        return annotationValues;
+    }
+
+    private static void extractedFieldName(Map<String, String> annotationValues, Map.Entry<BString, Object> qField,
+                                           BMap value) {
+        String[] parts = Pattern.compile(REGEX_FOR_FIELD).split(qField.getKey().getValue());
+        String fieldName = unescapeIdentifier(parts[1]);
+        Object overrideValue = value.get(HttpConstants.ANN_FIELD_NAME);
+        if (!(overrideValue instanceof BString overrideName)) {
+            return;
```

Comment:
Shall we add a test case where we set the `name: ()` and another test case without the name field?

Reply:
Added via bb44773